src/test/java/io/strimzi/kafka/bridge/mqtt/MqttBridgetIT.java

```diff
@@ -0,0 +1,252 @@
+
+        // instantiate the kafka consumer client
+        kafkaConsumerClient = new KafkaConsumer<>(
+                ImmutableMap.of(
+                        ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaBootstrapServers,
+                        ConsumerConfig.GROUP_ID_CONFIG, "gid-" + UUID.randomUUID(),
+                        ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, OffsetResetStrategy.EARLIEST.name().toLowerCase(Locale.ROOT)
+                ),
+                new StringDeserializer(),
+                new StringDeserializer()
+        );
+
+        // consumer client subscribes to the kafka topic
+        kafkaConsumerClient.subscribe(Collections.singletonList(KAFKA_TOPIC));
+
+        // start the bridge server in a separate thread
+        prepareServerThread = new Thread(() -> {
+            try {
+                EventLoopGroup bossGroup = new NioEventLoopGroup();
+                EventLoopGroup workerGroup = new NioEventLoopGroup();
+
+                BridgeConfig bridgeConfig = BridgeConfig.fromMap(config);
+
+                String mappingRulesPath = Objects.requireNonNull(MqttBridgetIT.class.getClassLoader().getResource("mapping-rules-regex.json")).getPath();
+                MappingRulesLoader.getInstance().init(mappingRulesPath);
+
+                mqttBridge = new MqttServer(bridgeConfig, bossGroup, workerGroup, ChannelOption.SO_KEEPALIVE);
+                mqttBridge.start();
+            } catch (InterruptedException interruptedException) {
+                // in this case, we are interrupting the thread, so we don't need to do anything
+                logger.info("Shutting down the server");
+            } catch (Exception e) {
+                logger.error("Exception occurred", e);
+            }
+        });
+        prepareServerThread.start();
+        Thread.sleep(5000);
+    }
+
+    /**
+     * Randomly create a new client id before each test
+     */
+    @BeforeEach
+    public void createClientId() {
+        mqttClientId = "mqtt-client-" + new Random().nextInt(20);
+    }
+
+    /**
+     * Test the client's connection to the Bridge
+     */
+    @Test
+    public void testConnection() throws MqttException {
+        try (MqttClient client = new MqttClient(MQTT_SERVER_URI, mqttClientId)) {
+            // Set up options for the connection
+            MqttConnectOptions options = new MqttConnectOptions();
+
+            IMqttToken conn = client.connectWithResult(options);
+
+            assertThat("The session present flag should be false",
+                    conn.getSessionPresent(), is(false));
+
+            assertThat("The connection return code should be 0",
+                    conn.getMessageId(), is(0));
+
+            assertThat("The client should be connected",
+                    client.isConnected(), is(true));
+
+            assertThat("The client id should be " + mqttClientId,
+                    client.getClientId(), is(mqttClientId));
```

Comment:
this check doesn't make much sense. It's not something you are getting back in the connection but you set it a few lines above on client creation. If you follow the constructor, it's just setting an internal member field. So you are testing that the constructor worked :-D Just remove it ;-)

Reply:
Agree